libs/design-system-components/src/components/ImageBlockGallery/image-block-grid-item.tsx

```diff
@@ -15,11 +20,27 @@ export interface IGridItemProps {
 }
 
 export const ImageBlockGridItem: React.FC<IGridItemProps> = props => {
-  const { image, images, gridStyle, aspectRatio = 'aspect-square', handleClickImage } = props;
+  const {
+    imageNotLoadedLabel,
+    image,
+    images,
+    gridStyle,
+    aspectRatio = 'aspect-square',
+    handleClickImage,
+  } = props;
 
   const imageSrc = React.useMemo(() => image, [image]);
 
   const [imgLoaded, setImgLoaded] = React.useState(false);
+  const [showImgFailedToLoad, setShowImgFailedToLoad] = React.useState(false);
+
+  React.useEffect(() => {
+    setTimeout(() => {
```

Comment:
you need to clear this timeout when the hook unmounts to avoid potential issues like set state after unmount.